influxdb3_write/src/write_buffer/mod.rs

```diff
@@ -793,6 +793,139 @@ mod tests {
         );
     }
 
+    #[tokio::test(flavor = "multi_thread", worker_threads = 2)]
+    async fn returns_chunks_across_parquet_and_buffered_data() {
+        let object_store: Arc<dyn ObjectStore> = Arc::new(InMemory::new());
+        let persister = Arc::new(PersisterImpl::new(Arc::clone(&object_store)));
+        let time_provider = Arc::new(MockProvider::new(Time::from_timestamp_nanos(0)));
+        let level_0_duration = Level0Duration::new_5m();
+        let write_buffer = WriteBufferImpl::new(
+            Arc::clone(&persister),
+            Arc::clone(&time_provider),
+            level_0_duration,
+            crate::test_help::make_exec(),
+            WalConfig {
+                level_0_duration: Duration::from_secs(60),
+                max_write_buffer_size: 100,
+                flush_interval: Duration::from_millis(10),
+                snapshot_size: 2,
+            },
+        )
+        .await
+        .unwrap();
+        let session_context = IOxSessionContext::with_testing();
+        let runtime_env = session_context.inner().runtime_env();
+        register_iox_object_store(runtime_env, "influxdb3", Arc::clone(&object_store));
```

Comment:
There is a `setup` helper below that does all of these steps, although, to use it here, looks like you would need to parameterize the different parts of the `WalConfig` in the function signature.

Reply:
Parameterized in 1eed007c4088b2055a6336dfea484f992452d5ce